webapp/src/components/cardDetail/cardDetailContents.test.tsx

```diff
@@ -0,0 +1,258 @@
+            const result = render(component)
+            container = result.container
+        })
+        expect(container).toMatchSnapshot()
+    })
+
+    test('should match snapshot with contents array', async () => {
+        const contents = [TestBlockFactory.createDivider(card)]
+        const component = wrapProviders((
+            <CardDetailContents
+                id='test-id'
+                card={card}
+                contents={contents}
+                readonly={false}
+            />
+        ))
+
+        let container: Element | undefined
+        await act(async () => {
+            const result = render(component)
+            container = result.container
+        })
+        expect(container).toMatchSnapshot()
+    })
+
+    test('should match snapshot after onBlur triggers', async () => {
+        const component = wrapProviders((
+            <CardDetailContents
+                id='test-id'
+                card={card}
+                contents={[]}
+                readonly={false}
+            />
+        ))
+
+        let container: Element | undefined
+        await act(async () => {
+            const result = render(component)
+            container = result.container
+        })
+        const markdownEditorField = container!.querySelectorAll('.octo-editor-preview.octo-placeholder')
+        expect(markdownEditorField).toBeDefined()
+        fireEvent.click(markdownEditorField[markdownEditorField.length - 1]!)
+
+        const onFocusEvent = new FocusEvent('focus', {
+            view: window,
+            bubbles: true,
+            cancelable: true,
+        })
+
+        const onBlurEvent = new FocusEvent('blur', {
+            view: window,
+            bubbles: true,
+            cancelable: true,
+        })
+
+        const textareaContainer = container!.querySelectorAll('.CodeMirror.cm-s-easymde.CodeMirror-wrap')
+        const textarea = textareaContainer[textareaContainer.length - 1].querySelector('textarea')
+
+        await act(async () => {
+            textarea!.dispatchEvent(onFocusEvent)
+            fireEvent.input(textarea!, {target: {value: 'test123'}})
+            fireEvent.keyPress(textarea!, {key: 'Escape', code: 'Escape'})
+            textarea!.dispatchEvent(onBlurEvent)
+        })
+
+        expect(container).toMatchSnapshot()
+    })
+
+    test('should match snapshot with contents array that has array inside it', async () => {
+        const contents = [TestBlockFactory.createDivider(card), [TestBlockFactory.createDivider(card), TestBlockFactory.createDivider(card)]]
+        const component = wrapProviders((
+            <CardDetailContents
+                id='test-id'
+                card={card}
+                contents={contents}
+                readonly={false}
+            />
+        ))
+
+        let container: Element | undefined
+        await act(async () => {
+            const result = render(component)
+            container = result.container
+        })
```

Comment:
Looks like this code is repeated in several tests. Does it make sense to extract it to function?

Reply:
I don't think so. Then that function would be repeated. I don't feel it's necessary to extract to a function here.